Sdk/MoyasarSdk/Services/PaymentServices/PaymentServices.swift

```diff
@@ -80,11 +72,39 @@ final public class PaymentService {
     
     public func createPayment(_ paymentRequest: ApiPaymentRequest) async throws -> ApiPayment {
         let payload = try encoder.encode(paymentRequest)
-        return try await makeRequest(createUrl, method: "POST", payload: payload)
+        let request = createPaymentRequest(payload: payload, url: createUrl)
+        return try await makeRequest(request)
     }
     
     public func createToken(_ tokenRequest: ApiTokenRequest) async throws -> ApiToken {
         let payload = try encoder.encode(tokenRequest)
-        return try await makeRequest(createTokenUrl, method: "POST", payload: payload)
+        let request = createPaymentRequest(payload: payload, url: createTokenUrl)
+        return try await makeRequest(request)
+    }
+    
+    private func createPaymentRequest(payload: Data, url: URL) -> URLRequest {
+        var request = URLRequest(url: url)
+        let auth = "\(apiKey):".data(using: .utf8)?.base64EncodedString()
+        request.setValue("Basic \(auth!)", forHTTPHeaderField: "Authorization")
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.setValue("moyasar-ios-sdk", forHTTPHeaderField: "X-MOYASAR-LIB")
+        request.httpMethod = "POST"
+        request.httpBody = payload
+        return request
+    }
+    
+    public func sendSTCPaymentRequest(url: URL, stcOtpRequest: StcOtpRequest) async throws -> ApiPayment {
+        let payload = try encoder.encode(stcOtpRequest)
+        let request = createStcPaymentRequest(payload: payload, url: url)
+        return try await makeRequest(request)
+    }
+    
+    private func createStcPaymentRequest(payload: Data, url: URL) -> URLRequest {
```

Comment:
Since all payments creation share the same scaffolding (like endpoint, headers, amount, description, etc) and only differs in the `source` type, please consider re-using the same `createPaymentRequest` method instead of creating a different method to handle STC Pay payment creation.

Reply:
i reuse same initialize payment yes for all kind of payments even in STC  , 
But the only change here in submit otp request  it need body otp_value  i send otp in body to be more safe way and i don't send apiKey in this case also i the url is not the same as create payment it is dynamice i got it from initialize payment response so i added separate function to create it's header to follow solid and make it easier for anyone to understand it.

Reply:
Oh I got confused by the method name! I thought `createStcPaymentRequest` will CREATE an STC Payment request, hence my original suggestion. 😅

Can you please consider renaming it to convey it's an STC Pay OTP verification request?